core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -3854,6 +3854,24 @@ private void checkLiteral2(String expression, String expected) {
     sql(expected).exec();
   }
 
+  @Test public void testCastInStringOperandOfComparison() {
+    final String query = "select \"employee_id\" "
+        + "from \"foodmart\".\"employee\" "
+        + "where 10 = cast('10' as int) and \"birth_date\" = cast('1914-02-02' as date) or "
+        + "\"hire_date\" = cast('1996-01-01 '||'00:00:00' as timestamp)";
+    final String expected = "SELECT \"employee_id\"\n"
+        + "FROM \"foodmart\".\"employee\"\n"
+        + "WHERE 10 = '10' AND \"birth_date\" = '1914-02-02' OR \"hire_date\" = '1996-01-01 ' || "
+        + "'00:00:00'";
+    final String expectedBiqquery = "SELECT employee_id\n"
+        + "FROM foodmart.employee\n"
+        + "WHERE 10 = CAST('10' AS INTEGER) AND birth_date = '1914-02-02' OR hire_date = "
+        + "CAST('1996-01-01 ' || '00:00:00' AS TIMESTAMP(0))";
```

Comment:
Honestly, I haven't seen `TIMESTAMP(0)` in BigQuery

Reply:
Should be fixed by CALCITE-3381 or a equivalent one

Reply:
Hi, I can see that CALCITE-3381 PR is merged. Resolving this conversion. Thanks for reviewing.